File: odemapster-core/src/main/java/es/upm/fi/dia/oeg/obdi/core/engine/RDBResultSet.java
package es.upm.fi.dia.oeg.obdi.core.engine;

import java.sql.ResultSet;
import java.sql.SQLException;
import es.upm.fi.dia.oeg.obdi.core.exception.ResultSetException;

public class RDBResultSet extends AbstractResultSet {
	private ResultSet rs;
	
	public RDBResultSet(ResultSet rs) {
		super();
		this.rs = rs;
	}
	
	public boolean next() throws ResultSetException {
		try {
			return this.rs.next();
		} catch (SQLException e) {
			return false;
		}
	}

	public Object getObject(int columnIndex) throws ResultSetException {
		try {
			return rs.getObject(columnIndex);
		} catch (SQLException e) {
			throw new ResultSetException(e);
		}
	}

	public Object getObject(String columnLabel) throws ResultSetException {
		try {
			return rs.getObject(columnLabel);
		} catch (SQLException e) {
			throw new ResultSetException(e);
		}
	}

	public String getString(int columnIndex) throws ResultSetException {
		try {
			return rs.getString(columnIndex);
		} catch (SQLException e) {
			throw new ResultSetException(e);
		}
	}

	public String getString(String columnLabel) throws ResultSetException {
		try {
			return rs.getString(columnLabel);
		} catch (SQLException e) {
			throw new ResultSetException(e);
		}
	}

	public Integer getInt(int columnIndex) throws ResultSetException {
		try {
			return rs.getInt(columnIndex);
		} catch (SQLException e) {
			throw new ResultSetException(e);
		}
	}

	public Integer getInt(String columnLabel) throws ResultSetException {
		try {
			return rs.getInt(columnLabel);
		} catch (SQLException e) {
			throw new ResultSetException(e);
		}
	}


}

File: morph-base/src/main/scala/es/upm/fi/dia/oeg/morph/base/engine/RDBResultSet.scala
package es.upm.fi.dia.oeg.morph.base.engine

import java.sql.ResultSet

class RDBResultSet(rs:ResultSet ) extends MorphBaseResultSet{
	def next() : Boolean = {
		try { this.rs.next(); } 
		catch  {
		  case e:Exception => { false; }
		}
	}

	def getObject(columnIndex:Int) : Object = {
	  rs.getObject(columnIndex);
	}

	def getObject(columnLabel:String ) : Object = {
	  rs.getObject(columnLabel);
	}

	def getString(columnIndex:Int ) : String  = {
		rs.getString(columnIndex);
	}

	def getString(columnLabel:String ) : String  = {
		rs.getString(columnLabel);
	}

	def getInt(columnIndex:Int ) : Integer = {
		return rs.getInt(columnIndex);
	}

	def getInt(columnLabel:String ) : Integer  = {
		rs.getInt(columnLabel);
	}
}